lib/index.js

```diff
@@ -19,6 +89,57 @@ function handleTokenRequest({logger}) {
     }
   };
 }
-odule.exports = {
-  handleTokenRequest
+
+function _authenticateClient({client, clientSecret}) {
```

Comment:
Should `clientSecret` be `expectedClientSecret` here?

Reply:
Yes.

Reply:
Er, no, wait. `clientSecret` is the secret incoming in the request. so it's not really expected.